xla/service/gpu/gpu_executable.cc

```diff
@@ -472,8 +472,17 @@ absl::Status ExecuteThunks(
         &collective_cliques};
 
     tsl::profiler::TraceMe trace([&] { return "Thunks::Initialize"; });
+    // cuDNN fusion thunks have to be initialized before they can be captured
+    // into command buffers at the initialization of command buffer thunks.
     for (const std::unique_ptr<Thunk>& thunk : thunk_sequence) {
-      TF_RETURN_IF_ERROR(thunk->Initialize(initialize_params));
+      if (thunk->kind() == Thunk::kCuDnn) {
```

Comment:
If state is properly shared we don't need any special cases here

Reply:
Done.